fix: Fallback Regex

rhit-daughewd opened the pull request.

Borketh reviewed (changes requested):
Could these safe regex functions be put in a new file in the `libraries` directory? Crashes is already a massive file. This way we can also use them elsewhere if needed. 
Please also run the black formatter.

Borketh commented:
closes #193

Borketh commented on fred/cogs/crashes.py:
```diff
@@ -553,14 +559,14 @@ def _get_fg_log_details(log_file: IO[bytes]):
+            r"Net CL: (?P<game_version>\d+)",
+            r"Command Line:(?P<cli>.*)",
+            r"Base Directory:(?P<path>.+)",
+            r"Launcher ID: (?P<launcher>\w+)",
```
Why avoid precompiling these?

rhit-daughewd commented:
I added back precompiling (I removed it when testing what an issue was and forgot to add it back, oops) and removed the 1-line commands. In doing so, I had to add the fallback setting to the other file. Currently, regex_util is the fallback for the timed out serach, and the rest are implemented in line in the 1-2 places that needed each.

Do we still want a regex_util file? In changing the other implementation, both files now set the fallback setting, so I could move the safe_search into crashes.py if it's no longer necessary.
